app/dao/unsubscribe_request_dao.py

```diff
@@ -10,3 +10,18 @@ def create_unsubscribe_request_dao(unsubscribe_data):
 
 def get_unsubscribe_request_by_notification_id_dao(notification_id):
     return UnsubscribeRequest.query.filter_by(notification_id=notification_id).one()
+
+
+def get_count_of_unprocessed_requests(notification_id):
+    return (
+        db.session.query(
+            UnsubscribeRequest.id.label("unsubscribe_request_id"),
+        )
+        .select_from(UnsubscribeRequest)
+        .join(
```

Comment:
join may not be needed in the first instance, check.

Filter on service_id in unsubscribe request table